src/transform.ts

```diff
@@ -46,6 +46,22 @@ export function addCoersion(def: any): void {
   }
 }
 
+export function findRefs(definition: any): string[] {
+  if (isPlainObject(definition)) {
+    const refs = flatMap(Object.values(definition), findRefs);
+    if (definition.$ref) {
+      return [definition.$ref, ...refs];
```

Comment:
Wouldn't `push`ing it at the end be considerably more efficient?

Reply:
Yes, but will also mutate the original array.